velox/functions/tests/FunctionRegistryTest.cpp

```diff
@@ -0,0 +1,312 @@
+    // map(K,V) -> array(K)
+    return {velox::exec::FunctionSignatureBuilder()
+                .typeVariable("K")
+                .typeVariable("V")
+                .returnType("array(K)")
+                .argumentType("map(K,V)")
+                .build()};
+  }
+};
+
+VELOX_DECLARE_VECTOR_FUNCTION(
+    udf_vector_func_one,
+    VectorFuncOne::signatures(),
+    std::make_unique<VectorFuncOne>());
+
+VELOX_DECLARE_VECTOR_FUNCTION(
+    udf_vector_func_two,
+    VectorFuncTwo::signatures(),
+    std::make_unique<VectorFuncTwo>());
+
+VELOX_DECLARE_VECTOR_FUNCTION(
+    udf_vector_func_three,
+    VectorFuncThree::signatures(),
+    std::make_unique<VectorFuncThree>());
+
+VELOX_DECLARE_VECTOR_FUNCTION(
+    udf_vector_func_four,
+    VectorFuncFour::signatures(),
+    std::make_unique<VectorFuncFour>());
+
+inline void registerTestFunctions() {
+  registerFunction<udf_func_one, Varchar, Varchar>({"func_one"});
+  registerFunction<udf_func_two<int64_t, int32_t>, int64_t, int64_t, int32_t>(
+      {"func_two"});
+  registerFunction<udf_func_two<int64_t, int16_t>, int64_t, int64_t, int16_t>(
+      {"func_two"});
+  registerFunction<udf_func_three, Array<int64_t>, Array<int64_t>>(
+      {"func_three"});
+
+  VELOX_REGISTER_VECTOR_FUNCTION(udf_vector_func_one, "vector_func_one");
+  VELOX_REGISTER_VECTOR_FUNCTION(udf_vector_func_two, "vector_func_two");
+  VELOX_REGISTER_VECTOR_FUNCTION(udf_vector_func_three, "vector_func_three");
+  VELOX_REGISTER_VECTOR_FUNCTION(udf_vector_func_four, "vector_func_four");
+}
+} // namespace
+
+class FunctionRegistryTest : public ::testing::Test {
+ public:
+  FunctionRegistryTest() {
+    registerTestFunctions();
+  }
+};
+
+TEST_F(FunctionRegistryTest, getFunctionSignatures) {
+  auto functionSignatures = getFunctionSignatures();
+  ASSERT_EQ(functionSignatures.size(), 7);
+
+  ASSERT_EQ(functionSignatures.count("func_one"), 1);
+  ASSERT_EQ(functionSignatures.count("func_two"), 1);
+  ASSERT_EQ(functionSignatures.count("func_three"), 1);
+  ASSERT_EQ(functionSignatures.count("vector_func_one"), 1);
+  ASSERT_EQ(functionSignatures.count("vector_func_two"), 1);
+  ASSERT_EQ(functionSignatures.count("vector_func_three"), 1);
+  ASSERT_EQ(functionSignatures.count("vector_func_four"), 1);
+
+  ASSERT_EQ(functionSignatures["func_one"].size(), 1);
+  ASSERT_EQ(functionSignatures["func_two"].size(), 2);
+  ASSERT_EQ(functionSignatures["func_three"].size(), 1);
+  ASSERT_EQ(functionSignatures["vector_func_one"].size(), 1);
+  ASSERT_EQ(functionSignatures["vector_func_two"].size(), 1);
+  ASSERT_EQ(functionSignatures["vector_func_three"].size(), 1);
+  ASSERT_EQ(functionSignatures["vector_func_four"].size(), 1);
+
+  ASSERT_EQ(
+      functionSignatures["func_one"].at(0)->toString(),
+      exec::FunctionSignatureBuilder()
+          .returnType("varchar")
+          .argumentType("varchar")
+          .build()
+          ->toString());
+
+  ASSERT_EQ(
+      functionSignatures["func_two"].at(0)->toString(),
```

Comment:
Looks like the order of signatures here is unspecified and this test can randomly fail.